server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -152,6 +157,52 @@
 public class CompactionCoordinator
     implements CompactionCoordinatorService.Iface, Runnable, MetricsProducer {
 
+  // Object that serves as a TopN view of the RunningCompactions, ordered by
+  // RunningCompaction start time. The first entry in this Set should be the
+  // oldest RunningCompaction.
+  private static class TimeOrderedRunningCompactionSet
+      extends ConcurrentSkipListSet<RunningCompaction> {
+
+    // TODO: Could make this configurable, but probably should not be large. These
+    // are shown on the monitor, can't imagine that someone is going to page through
+    // all of them.
+    private static final int UPPER_LIMIT = 250;
```

Comment:
I agree it should not be large. I would maybe make it smaller, whatever will comfortably fit on a single web page.